oppia_android_test.bzl

```diff
@@ -0,0 +1,45 @@
+load("@rules_jvm_external//:defs.bzl", "artifact")
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_android_library")
+
+def oppia_android_test(name, srcs, test_manifest, custom_package, resource_files,
+                       test_class, src_library_name):
+  '''
+  This macro exists as a way to set up a test in Oppia Android to be run with Bazel.
+  This macro creates a library for an individual test that is fed as a dependency into an
+  android_local_test() rule which configures the test to be run with Roboletric and Bazel.
+
+  Args:
+      name: str. The name of the Kotlin test file without the '.kt' suffix.
+      srcs: str. The name of the Kotlin test file to be run.
+      test_manifest: str. The path to the test manifest file.
+      custom_package: str. The module's package. Example: 'org.oppia.utility'.
+      resource_files: str. The path to the resource files. This is typically a glob([]).
+      test_class: The package of the src file. For example, if the src is 'FakeEventLoggerTest.kt',
+          then the test_class would be "org.oppia.testing.FakeEventLoggerTest".
+      src_library_name: str. The name of the library that builds the module's test files.
```

Comment:
Why do we need this? Can't we just pass in the deps that the test needs & build each test by itself via its test library below? The additional combined tests libraries seem unnecessary.

I was assuming that this macro would essentially be a final test macro we would use into & post phase 2, whereas the separate ``domain_test``-esque macros are temporary in that they simplify the test definitions (maybe even sharing a deps list).

Reply:
You are right! `src_library_name` is unnecessary! As for the dependencies, are you saying that macros like `testing_test()` and `domain_test()` should feed in module-specific dependencies into `oppia_android_test` or are you thinking that the actual calls to `testing_test()` and `domain_test()` found in the BUILD file should feed in test-specific dependencies? From what I have seen, most of the time dependencies don't vary too much from test to test within a module... but what do you think? Maybe there is value is having the list of dependencies visible from the BUILD file (in terms of readability, maybe?). For now, I have just created a general TEST_DEPS list in the BUILD file.

Reply:
I see you chose to do the second & define the test deps at the test definitions. I think that's fine--I had the former in mind (to share the same deps list for all tests of a particular category), but both work and are reasonable. Phase 2 will have specific deps lists for each test, so both solutions will be replaced in the coming weeks.